From 0d88b73d189ea3440ccaab00418d6469f76fa590 Mon Sep 17 00:00:00 2001
From: Michal Hocko <mhocko@suse.com>
Date: Wed, 18 Jul 2018 11:24:29 +0200
Subject: [PATCH] cgrulesengd: remove umask(0)

One of our partners has noticed that cgred daemon is creating a log file
(/var/log/cgred) with too wide permissions (0666) and that is seen as
a security bug because an untrusted user can write to otherwise
restricted area. CVE-2018-14348 has been assigned to this issue.

Signed-off-by: Michal Hocko <mhocko@suse.com>
Acked-by: Balbir Singh <bsingharora@gmail.com>
[Retrieved from:
https://github.com/libcgroup/libcgroup/commit/0d88b73d189ea3440ccaab00418d6469f76fa590]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 src/daemon/cgrulesengd.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/daemon/cgrulesengd.c b/src/daemon/cgrulesengd.c
index ea51f11..0d288f3 100644
--- a/src/daemon/cgrulesengd.c
+++ b/src/daemon/cgrulesengd.c
@@ -889,9 +889,6 @@ int cgre_start_daemon(const char *logp, const int logf,
 		} else if (pid > 0) {
 			exit(EXIT_SUCCESS);
 		}
-
-		/* Change the file mode mask. */
-		umask(0);
 	} else {
 		flog(LOG_DEBUG, "Not using daemon mode\n");
 		pid = getpid();
